install-and-configure/install/multi-cluster/federated-etl/thanos-migration-guide.md

```diff
@@ -0,0 +1,155 @@
+
+### Step 1: Use the existing Thanos object store or create a new dedicated object store
+
+This object store is where the ETL backups will be pushed to from the primary cluster.
+
+### Step 2: Enable [ETL Backups](https://docs.kubecost.com/install-and-configure/install/etl-backup#google-cloud-storage) on the *primary cluster only*. 
+
+This will ensure we have all historical data in durable storage and it will create the directory the ETL Utils container needs in order to create the directory structure needed for Aggregator to pull the ETL from each respective cluster.
+
+```
+kubecostModel:
+  etlBucketConfigSecret: <YOUR_SECRET_NAME>
+```
+
+### Step 3: Validate that an `/etl` directory is present in the object store
+
+There should be ETL data present in the following directories. CloudCosts will only have ETL data if cloud integration is enabled.
+
+* `/etl/bingen/allocations`
+* `/etl/bingen/assets/`
+* `/cloudcosts/`
+
+
+### Step 4: Create a new secret for the federated store
+
+This will point to the existing Thanos object store or the new one created in Step 1.
+
+{% hint style="warning" %}
+The name of the .yaml file used to create the secret *must* be named _federated-store.yaml_ or Aggregator will not start.
+{% endhint %}
+
+
+```
+kubectl create secret generic federated-store --from-file=federated-store.yaml -n kubecost
+```
+
+### Step 5: Set Aggregator and ETL Utils in *values.yaml* on the primary cluster.
+
+{% hint style="warning" %}
+*Do not* disable Thanos during this step. This will not negatively impact the source data. Thanos will be disabled in a later step.
+{% endhint %}
+
+* [Enable Aggregator for the primary cluster](https://docs.kubecost.com/install-and-configure/install/multi-cluster/federated-etl/aggregator). 
+
+Note:  
+
+* ETL Utils
+```
+etlUtils:
+  thanosSourceBucketSecret: kubecost-thanos
+  affinity: {}
+  enabled: true
+  env:
+    LOG_LEVEL: debug
+  fullImageName: gcr.io/kubecost1/cost-model-etl-utils:0.2
+  nodeSelector:
+    kubernetes.io/arch: amd64
+  resources: {}
+  tolerations: {}
+kubecostModel:
+  federatedStorageConfigSecret: kubecost-thanos
+```
+
+### Step 6: Apply the changes and wait for data to populate in the UI. 
+
+Ensure all pods and containers are running:
+
+ ```
+kubecost-cost-analyzer-685fd8f677-k652h        4/4     Running   2 (12m ago)   3h2m
+kubecost-forecasting-6cf4dd7b98-7z8f5          1/1     Running   0             66m
+kubecost-etl-utils-6cdd489596-5dl75           1/1     Running   0          6d20h
+```
+
+This can take some time depending on how many unique containers are running in the environment and how much ETL data is in the object store. A couple importants steps are happening in the background which take time to complete.
+
+* The ETL Utils image is building the directory structure in the object store needed by Aggregator to pull the ETL data. 
+* SQL tables are building.
```

Comment:
What is this supposed to mean?

Reply:
@michaelmdresser - attempted to come up with some wording around what's happening in the background once aggregator and etl utils is deployed. My understanding and experience is it takes some time for data to come up in the UI. I think adding a banner in the UI letting the user know to wait would be ideal.

Reply:
Got it, I think we're testing some approaches to such a banner in eng right now -- I'm not sure if that will make 2.0 yet but its definitely a priority.